src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -877,7 +880,7 @@ private void CheckNewModifier(Symbol symbol, bool isNew, DiagnosticBag diagnosti
                                 diagnostics.Add(ErrorCode.ERR_CantChangeRefReturnOnOverride, overridingMemberLocation, overridingMember, overriddenMember);
                                 suppressAccessors = true; //we get really unhelpful errors from the accessor if the ref kind is mismatched
                             }
-                            else if (!overridingMemberType.Equals(overriddenMemberType, TypeCompareKind.AllIgnoreOptions))
+                            else if (!IsValidOverrideReturnType(overridingProperty, overridingMemberType, overriddenMemberType, diagnostics))
                             {
                                 // if the type is or contains an error type, the type must be fixed before the override can be found, so suppress error
                                 if (!isOrContainsErrorType(overridingMemberType.Type))
```

Comment:
>if (!isOrContainsErrorType(overridingMemberType.Type)) [](start = 32, length = 54)

It looks like we are not reporting language version mismatch on this code path. Is the assumption that this would be covered by the getter? Consider adding a comment about that.

Reply:
This appears to be addressed in [https://github.com/dotnet/roslyn/pull/43673](https://github.com/dotnet/roslyn/pull/43673) iteration 1.

---
In reply to: [416157623](https://github.com/dotnet/roslyn/pull/43576#discussion_r416157623) [](ancestors = 416157623)